[2021-10-30 13:42:40,352] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: airflow-cleanup-logs.log_cleanup_worker_num_1_dir_0 2021-10-30T13:42:36.608966+00:00 [queued]>
[2021-10-30 13:42:40,415] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: airflow-cleanup-logs.log_cleanup_worker_num_1_dir_0 2021-10-30T13:42:36.608966+00:00 [queued]>
[2021-10-30 13:42:40,418] {taskinstance.py:1042} INFO - 
--------------------------------------------------------------------------------
[2021-10-30 13:42:40,419] {taskinstance.py:1043} INFO - Starting attempt 1 of 2
[2021-10-30 13:42:40,420] {taskinstance.py:1044} INFO - 
--------------------------------------------------------------------------------
[2021-10-30 13:42:40,452] {taskinstance.py:1063} INFO - Executing <Task(BashOperator): log_cleanup_worker_num_1_dir_0> on 2021-10-30T13:42:36.608966+00:00
[2021-10-30 13:42:40,457] {standard_task_runner.py:52} INFO - Started process 4049 to run task
[2021-10-30 13:42:40,468] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'airflow-cleanup-logs', 'log_cleanup_worker_num_1_dir_0', '2021-10-30T13:42:36.608966+00:00', '--job-id', '54891', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/airflow-cleanup-logs.py', '--cfg-path', '/tmp/tmp8ygovlh1', '--error-file', '/tmp/tmp_z_9j1nd']
[2021-10-30 13:42:40,469] {standard_task_runner.py:77} INFO - Job 54891: Subtask log_cleanup_worker_num_1_dir_0
[2021-10-30 13:42:40,605] {logging_mixin.py:104} INFO - Running <TaskInstance: airflow-cleanup-logs.log_cleanup_worker_num_1_dir_0 2021-10-30T13:42:36.608966+00:00 [running]> on host f8e9f518f089
[2021-10-30 13:42:40,715] {taskinstance.py:1257} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=bwiratmo91@gmail.com
AIRFLOW_CTX_DAG_OWNER=bayu
AIRFLOW_CTX_DAG_ID=airflow-cleanup-logs
AIRFLOW_CTX_TASK_ID=log_cleanup_worker_num_1_dir_0
AIRFLOW_CTX_EXECUTION_DATE=2021-10-30T13:42:36.608966+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2021-10-30T13:42:36.608966+00:00
[2021-10-30 13:42:40,718] {bash.py:135} INFO - Tmp dir root location: 
 /tmp
[2021-10-30 13:42:40,720] {bash.py:158} INFO - Running command: 

echo "Getting Configurations..."
BASE_LOG_FOLDER="/opt/airflow/logs"
WORKER_SLEEP_TIME="3"

sleep ${WORKER_SLEEP_TIME}s

MAX_LOG_AGE_IN_DAYS=""
if [ "${MAX_LOG_AGE_IN_DAYS}" == "" ]; then
    echo "maxLogAgeInDays conf variable isn't included. Using Default '1'."
    MAX_LOG_AGE_IN_DAYS='1'
fi
ENABLE_DELETE=true
echo "Finished Getting Configurations"
echo ""

echo "Configurations:"
echo "BASE_LOG_FOLDER:      '${BASE_LOG_FOLDER}'"
echo "MAX_LOG_AGE_IN_DAYS:  '${MAX_LOG_AGE_IN_DAYS}'"
echo "ENABLE_DELETE:        '${ENABLE_DELETE}'"

cleanup() {
    echo "Executing Find Statement: $1"
    FILES_MARKED_FOR_DELETE=`eval $1`
    echo "Process will be Deleting the following File(s)/Directory(s):"
    echo "${FILES_MARKED_FOR_DELETE}"
    echo "Process will be Deleting `echo "${FILES_MARKED_FOR_DELETE}" |     grep -v '^$' | wc -l` File(s)/Directory(s)"         # "grep -v '^$'" - removes empty lines.
    # "wc -l" - Counts the number of lines
    echo ""
    if [ "${ENABLE_DELETE}" == "true" ];
    then
        if [ "${FILES_MARKED_FOR_DELETE}" != "" ];
        then
            echo "Executing Delete Statement: $2"
            eval $2
            DELETE_STMT_EXIT_CODE=$?
            if [ "${DELETE_STMT_EXIT_CODE}" != "0" ]; then
                echo "Delete process failed with exit code                     '${DELETE_STMT_EXIT_CODE}'"

                echo "Removing lock file..."
                rm -f /tmp/airflow_log_cleanup_worker.lock
                if [ "${REMOVE_LOCK_FILE_EXIT_CODE}" != "0" ]; then
                    echo "Error removing the lock file.                     Check file permissions.                    To re-run the DAG, ensure that the lock file has been                     deleted (/tmp/airflow_log_cleanup_worker.lock)."
                    exit ${REMOVE_LOCK_FILE_EXIT_CODE}
                fi
                exit ${DELETE_STMT_EXIT_CODE}
            fi
        else
            echo "WARN: No File(s)/Directory(s) to Delete"
        fi
    else
        echo "WARN: You're opted to skip deleting the File(s)/Directory(s)!!!"
    fi
}


if [ ! -f /tmp/airflow_log_cleanup_worker.lock ]; then

    echo "Lock file not found on this node!     Creating it to prevent collisions..."
    touch /tmp/airflow_log_cleanup_worker.lock
    CREATE_LOCK_FILE_EXIT_CODE=$?
    if [ "${CREATE_LOCK_FILE_EXIT_CODE}" != "0" ]; then
        echo "Error creating the lock file.         Check if the airflow user can create files under tmp directory.         Exiting..."
        exit ${CREATE_LOCK_FILE_EXIT_CODE}
    fi

    echo ""
    echo "Running Cleanup Process..."

    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/*/* -type f -mtime      +${MAX_LOG_AGE_IN_DAYS}"
    DELETE_STMT="${FIND_STATEMENT} -exec rm -f {} \;"

    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
    CLEANUP_EXIT_CODE=$?

    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/*/* -type d -empty"
    DELETE_STMT="${FIND_STATEMENT} -prune -exec rm -rf {} \;"

    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
    CLEANUP_EXIT_CODE=$?

    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/* -type d -empty"
    DELETE_STMT="${FIND_STATEMENT} -prune -exec rm -rf {} \;"

    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
    CLEANUP_EXIT_CODE=$?

    echo "Finished Running Cleanup Process"

    echo "Deleting lock file..."
    rm -f /tmp/airflow_log_cleanup_worker.lock
    REMOVE_LOCK_FILE_EXIT_CODE=$?
    if [ "${REMOVE_LOCK_FILE_EXIT_CODE}" != "0" ]; then
        echo "Error removing the lock file. Check file permissions. To re-run the DAG, ensure that the lock file has been deleted (/tmp/airflow_log_cleanup_worker.lock)."
        exit ${REMOVE_LOCK_FILE_EXIT_CODE}
    fi

else
    echo "Another task is already deleting logs on this worker node.     Skipping it!"
    echo "If you believe you're receiving this message in error, kindly check     if /tmp/airflow_log_cleanup_worker.lock exists and delete it."
    exit 0
fi

[2021-10-30 13:42:40,742] {bash.py:169} INFO - Output:
[2021-10-30 13:42:40,746] {bash.py:173} INFO - Getting Configurations...
[2021-10-30 13:42:43,750] {bash.py:173} INFO - maxLogAgeInDays conf variable isn't included. Using Default '1'.
[2021-10-30 13:42:43,763] {bash.py:173} INFO - Finished Getting Configurations
[2021-10-30 13:42:43,765] {bash.py:173} INFO - 
[2021-10-30 13:42:43,765] {bash.py:173} INFO - Configurations:
[2021-10-30 13:42:43,768] {bash.py:173} INFO - BASE_LOG_FOLDER:      '/opt/airflow/logs'
[2021-10-30 13:42:43,769] {bash.py:173} INFO - MAX_LOG_AGE_IN_DAYS:  '1'
[2021-10-30 13:42:43,775] {bash.py:173} INFO - ENABLE_DELETE:        'true'
[2021-10-30 13:42:43,777] {bash.py:173} INFO - Lock file not found on this node!     Creating it to prevent collisions...
[2021-10-30 13:42:43,778] {bash.py:173} INFO - 
[2021-10-30 13:42:43,787] {bash.py:173} INFO - Running Cleanup Process...
[2021-10-30 13:42:43,788] {bash.py:173} INFO - Executing Find Statement: find /opt/airflow/logs/*/* -type f -mtime      +1
[2021-10-30 13:44:07,180] {bash.py:173} INFO - Process will be Deleting the following File(s)/Directory(s):
[2021-10-30 13:44:07,180] {bash.py:173} INFO - 
[2021-10-30 13:44:07,182] {bash.py:173} INFO - Process will be Deleting 0 File(s)/Directory(s)
[2021-10-30 13:44:07,182] {bash.py:173} INFO - 
[2021-10-30 13:44:07,182] {bash.py:173} INFO - WARN: No File(s)/Directory(s) to Delete
[2021-10-30 13:44:07,182] {bash.py:173} INFO - Executing Find Statement: find /opt/airflow/logs/*/* -type d -empty
[2021-10-30 13:45:37,512] {bash.py:173} INFO - Process will be Deleting the following File(s)/Directory(s):
[2021-10-30 13:45:37,513] {bash.py:173} INFO - 
[2021-10-30 13:45:37,516] {bash.py:173} INFO - Process will be Deleting 0 File(s)/Directory(s)
[2021-10-30 13:45:37,519] {bash.py:173} INFO - 
[2021-10-30 13:45:37,521] {bash.py:173} INFO - WARN: No File(s)/Directory(s) to Delete
[2021-10-30 13:45:37,522] {bash.py:173} INFO - Executing Find Statement: find /opt/airflow/logs/* -type d -empty
[2021-10-30 13:47:14,977] {bash.py:173} INFO - Process will be Deleting the following File(s)/Directory(s):
[2021-10-30 13:47:14,978] {bash.py:173} INFO - 
[2021-10-30 13:47:14,981] {bash.py:173} INFO - Process will be Deleting 0 File(s)/Directory(s)
[2021-10-30 13:47:14,982] {bash.py:173} INFO - 
[2021-10-30 13:47:14,982] {bash.py:173} INFO - WARN: No File(s)/Directory(s) to Delete
[2021-10-30 13:47:14,982] {bash.py:173} INFO - Finished Running Cleanup Process
[2021-10-30 13:47:14,983] {bash.py:173} INFO - Deleting lock file...
[2021-10-30 13:47:14,984] {bash.py:177} INFO - Command exited with return code 0
[2021-10-30 13:47:15,029] {taskinstance.py:1166} INFO - Marking task as SUCCESS. dag_id=airflow-cleanup-logs, task_id=log_cleanup_worker_num_1_dir_0, execution_date=20211030T134236, start_date=20211030T134240, end_date=20211030T134715
[2021-10-30 13:47:15,206] {local_task_job.py:188} WARNING - State of this instance has been externally set to success. Terminating instance.
[2021-10-30 13:47:15,217] {process_utils.py:100} INFO - Sending Signals.SIGTERM to GPID 4049
[2021-10-30 13:47:15,696] {taskinstance.py:1239} ERROR - Received SIGTERM. Terminating subprocesses.
[2021-10-30 13:47:15,696] {bash.py:185} INFO - Sending SIGTERM signal to bash process group
[2021-10-30 13:47:15,723] {process_utils.py:66} INFO - Process psutil.Process(pid=4049, status='terminated', exitcode=1, started='13:42:39') (4049) terminated with exit code 1
